MCprep_addon/spawner/mcmodel.py

```diff
@@ -0,0 +1,256 @@
+        rotate_around(rot_angle,[elm_to[0], elm_from[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_from[1], elm_from[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_to[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_to[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_to[0], elm_from[1], elm_to[2]],rot_origin,rot_axis),
+        rotate_around(rot_angle,[elm_from[0], elm_from[1], elm_to[2]],rot_origin,rot_axis),
+    ]
+
+    edges = []
+    faces = [[0, 1, 2, 3],[5, 4, 7, 6],[1, 0, 4, 5],[7, 6, 2, 3],[4, 0, 3, 7],[1, 5, 6, 2]]
+    
+    return [verts,edges,faces]
+
+def add_material(name="material", pth=""): # creates a simple material with an image texture from pth
+    conf.log(name+": "+pth)
+    mat = bpy.data.materials.new(name=name)
+    mat.blend_method = 'CLIP'
+    mat.shadow_method = 'CLIP'
+    mat.use_nodes = True
+    matnodes = mat.node_tree.nodes
+
+    tex = matnodes.new('ShaderNodeTexImage')
+    if os.path.isfile(pth):
+        img = bpy.data.images.load(pth, check_existing=True)
+        tex.image=img
+    tex.interpolation='Closest'
+
+    shader = matnodes['Principled BSDF']
+    mat.node_tree.links.new(shader.inputs['Base Color'], tex.outputs['Color'])
+    mat.node_tree.links.new(shader.inputs['Alpha'], tex.outputs['Alpha'])
+
+    return mat
+
+def locate_image(textures,img,model_filepath):
+
+    context=bpy.context
+    resource_folder = bpy.path.abspath(context.scene.mcprep_texturepack_path)
+
+    local_path = textures[img]
+    if local_path[0] == '#': #reference to another texture
+        return locate_image(textures,local_path[1:],model_filepath)
+    else:
+        if local_path[0] == '.': #path is local to the model file
+            directory = os.path.dirname(model_filepath)
+        else:
+            if(len(local_path.split(":"))==1):
+                namespace="minecraft"
+            else:
+                namespace=local_path.split(":")[0]
+                local_path=local_path.split(":")[1]
+
+            directory = os.path.join(resource_folder,"assets",namespace,"textures")
+        return os.path.realpath(os.path.join(directory,local_path)+".png")
+
+def read_model(model_filepath): # recursive function that extracts the information from the json files
+    with open(model_filepath, 'r') as f:
+            obj_data=json.loads(f.read())
+    
+    context=bpy.context
+    resource_folder = bpy.path.abspath(context.scene.mcprep_texturepack_path)
+    
+    elements=None
+    textures=None
+
+    parent = obj_data.get("parent")
+    if parent is not None:
+        if parent=="builtin/generated" or parent=="item/generated":
+            pass # generates the model from the texture
+        elif parent=="builtin/entity":
+            pass # model from an entity file, only for chests, ender chests, mob heads, shields, banners and tridents
```

Comment:
Thanks for the comment details here! Is this list exhaustive, or are there some other examples too? (if others, just end it with `, etc.`)